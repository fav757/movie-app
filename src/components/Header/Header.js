import React from 'react';
import Logotype from '../Logotype/Logotype';
import Navigation from '../Navigation/Navigation';
import Search from '../Search/Search';
import styles from './Header.module.scss';

function Header() {
  return (
    <header className={styles.header}>
      <div className={styles.container}>
        <Logotype />
        <Navigation />
        <Search />
      </div>
    </header>
  );
}

export default Header;
