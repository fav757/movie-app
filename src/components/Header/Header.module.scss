.header {
  background: #080B0F;
  color: white;
}

.container {
  display: flex;
  align-items: center;
  margin: auto;
  max-width: 1200px;
}