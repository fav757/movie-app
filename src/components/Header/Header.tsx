import React, { useCallback, useState } from 'react';
import Logotype from '../Logotype/Logotype';
import Navigation from '../Navigation/Navigation';
import Search from '../Search/Search';
import GuestCreator from '../GuestCreator/GuestCreator';
import styles from './Header.module.scss';

const Header: React.FC = () => {
  const [showMenu, setShowMenu] = useState(false);
  const handleClick = useCallback(() => setShowMenu((state) => !state), []);
  const closeModal = () => setShowMenu(false);

  return (
    <header className={styles.container}>
      <div className={styles.header}>
        <Logotype />
        <button
          type="button"
          onClick={handleClick}
          aria-label="burger menu"
          className={`${styles.burger} fas fa-bars`}
        />
        <div className={`${styles.menu} ${showMenu || styles.hidden}`}>
          <Navigation closeModal={closeModal} />
          <Search closeModal={closeModal} />
          <GuestCreator closeModal={closeModal} />
        </div>
      </div>
    </header>
  );
};

export default Header;
