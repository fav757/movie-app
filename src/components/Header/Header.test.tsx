import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from './Header';

jest.mock('../Logotype/Logotype', () => () => <i data-testid="Logotype" />);
jest.mock('../Search/Search.tsx', () => () => <i data-testid="Search" />);
jest.mock('../Navigation/Navigation.tsx', () => () => (
  <i data-testid="Navigation" />
));

describe('Header component', () => {
  test('should render logotype, navigation and search', () => {
    render(<Header />);
    expect(screen.getByTestId('Logotype')).toBeInTheDocument();
    expect(screen.getByTestId('Search')).toBeInTheDocument();
    expect(screen.getByTestId('Navigation')).toBeInTheDocument();
  });

  test('should render logotype, navigation and search', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
