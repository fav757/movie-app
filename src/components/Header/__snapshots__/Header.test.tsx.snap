// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render logotype, navigation and search 1`] = `
<div>
  <header
    class="container"
  >
    <div
      class="header"
    >
      <i
        data-testid="Logotype"
      />
      <button
        aria-label="burger menu"
        class="burger fas fa-bars"
        type="button"
      />
      <div
        class="menu hidden"
      >
        <i
          data-testid="Navigation"
        />
        <i
          data-testid="Search"
        />
        <i
          data-testid="Guest creator"
        />
      </div>
    </div>
  </header>
</div>
`;
